doc/security/sensor-threat.rst

```diff
@@ -0,0 +1,327 @@
+      application image as well as updating the application image from
+      the update image when an update is needed.
+
+      The bootloader shall verify the signature of the update image
+      before installing it.
+
+      The bootloader shall only accept an update image with a newer
+      version number than the current image.
+
+   b. **The application image**. The application image contains the code
+      executed during normal operation of the device. Before running
+      this image, the bootloader shall verify a digital signature of the
+      image, to avoid running an image that has been tampered with. The
+      flash/system shall be configured such that after the bootloader
+      has completed, the CPU will be unable to write to the application
+      image.
+
+   c. **The update image**. This is an area of flash that holds a new
+      version of the application image. This image will be downloaded
+      and stored by the application during normal operation. When this
+      has completed, the application can cause a reboot, and the
+      bootloader can install the new image.
+
+   d. **Secret storage**. An area of the flash will be used to store
+      client secrets. This area is written and read by a subset of the
+      application image. The application shall be configured to
+      protect this area from both reads and writes by code that does
+      not need to have access to it, giving consideration to possible
+      exploits found within a majority of the application code.
+      Revealing the contents of the secrets would allow the attacker
+      to spoof this device.
+
+      Initial secrets shall be placed in the device during a
+      provisioning activity, distinct from normal operation of the
+      device. Later updates can be made under the direction of
+      communication received over a secured channel to the service.
+
+   e. **Configuration storage**. There shall be an area to store other
+      configuration information. On resource-constrained devices, it is
+      allowed for this to be stored in the same region as the secret
+      storage, however, this adds additional code that has access to the
+      secret storage area, and as such, more code that must be
+      scrutinized.
+
+   f. **Log storage**. The device may have an area of flash where log
+      events can be written.
+
+2. **Sensor/Actuator interface**. In this design, the sensor or actuator
+   communicates with the SoC via a bus, such as SPI. The hardware design
+   shall be made to make intercepting this bus difficult for an attack.
+   Required techniques depend on the sensitivity and use of the sensor
+   data, and can range from having the sensor mounted on the same PCB as
+   the MCU to expoxy potting the entire device.
+
+3. **Communication with cloud service**. Communication between the
+   device, and the cloud service will be done over the general
+   internet. As such, it shall be assumed that an attacker is able to
```

Comment:
A bit awkward phrasing.  Does this work:

    As such, it shall be assumed that an attacker can
    arbitrarily intercept this channel and, for example, return spoofed
    DNS results or attempt man-in-the-middle attacks
    against communication with cloud services.